src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -171,7 +183,80 @@ def acr_update_set(client,
                    registry_name,
                    resource_group_name=None,
                    parameters=None):
-    if resource_group_name is None:
-        resource_group_name = get_resource_group_name_by_registry_name(registry_name)
+    """Sets the properties of the specified container registry.
+    :param str registry_name: The name of container registry
+    :param str resource_group_name: The name of resource group
+    :param RegistryUpdateParameters parameters: The registry update parameters object
+    """
+    registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
+
+    if parameters.storage_account is not None and registry.sku.name != SkuTier.basic.value:  # pylint: disable=no-member
```

Comment:
The truthiness of an empty string is __False__. Both None and empty in my opinion should render the same failed result.